lua/glance/init.lua

```diff
@@ -138,11 +138,15 @@ local function open(opts)
 
   is_fetching = true
   local lsp = require('glance.lsp')
+  local request_bufnr = vim.api.nvim_get_current_buf()
+  local params = vim.lsp.util.make_position_params()
+  if is_open() then
+    Glance.actions.close()
+  end
   local parent_bufnr = vim.api.nvim_get_current_buf()
   local parent_winnr = vim.api.nvim_get_current_win()
-  local params = vim.lsp.util.make_position_params()
 
```

Comment:
If the `request_bufnr` is a different buffer, we need to load it and open it in the parent window.
```lua
  if request_bufnr ~= parent_bufnr then
    -- after we closed the glance window, all the preview buffers are unloaded
    -- we need to reopen the requested buffer
    request_bufnr = vim.uri_to_bufnr(params.textDocument.uri)

    if vim.fn.buflisted(request_bufnr) ~= 1 then
      vim.api.nvim_buf_set_option(request_bufnr, 'buflisted', true)
    end

    vim.api.nvim_win_set_buf(parent_winnr, request_bufnr)
    parent_bufnr = request_bufnr
  end
```
something like that should go before sending the lsp request.

Reply:
I wouldn't want to change the "real" buffer in any way. For me this feature is about "drilling down" into a type calling `Glance definiton` over and over again (for example this can be useful in TypeScript for complex nested parameter objects)

Reply:
You didn't get my point, your solution only works if the requested buffer has already been listed and loaded in your editor before calling glance, if you try to call it on a preview buffer you haven't opened before - glance will just break.
```lua
local request_bufnr = vim.api.nvim_get_current_buf()
if is_open() then
  Glance.actions.close()
end

```
When you close glance, all the preview buffers are unloaded if they haven't been listed before outside of glance, so `request_bufnr` most likely is no longer a valid buffer which will break glance execution.
That snippet i provided earlier solves that by loading the requested buffer

Reply:
I understand that we need to make sure that the buffer corresponding to `request_bufnr` is still loader, but I don't understand why we'd need to list it and load it into a window. In my usecase I'm calling glance again because the first result doesn't have the information I need.
Is it because otherwise these buffers would never be closed?

Reply:
because after closing and re-opening glance it will position the window according to `make_position_params` method result:
```lua
local params = vim.lsp.util.make_position_params()
```
if we keep the parent buffer, it will position the window in a random place.

I thought of another way of doing it:
```lua
  local request_bufnr = vim.api.nvim_get_current_buf()
  local params = vim.lsp.util.make_position_params()
  local parent_params = params

  if is_open() then
    Glance.actions.close()
    -- save the params for the parent window
    parent_params = vim.lsp.util.make_position_params()
  end

  local parent_bufnr = vim.api.nvim_get_current_buf()
  local parent_winnr = vim.api.nvim_get_current_win()

  if not vim.api.nvim_buf_is_loaded(request_bufnr) then
    -- after we closed the glance window, all the preview buffers are unloaded
    -- we need to reopen the buffer
    request_bufnr = vim.uri_to_bufnr(params.textDocument.uri)
    vim.fn.bufload(request_bufnr)
  end

  lsp.request(opts.method, params, request_bufnr, function(results, ctx)
```
and later
```lua
-- use parent_params instead of params for positioning
create(_results, parent_bufnr, parent_winnr, parent_params, opts.method)
```
this way we can keep the original position of the window and keep the parent buffer the same.
Could you please test it?